lib/containers/container_images.pm

```diff
@@ -85,7 +87,13 @@ sub build_with_zypper_docker {
 
 
     record_info("Testing derived");
-    test_opensuse_based_image(image => $derived_image, runtime => $runtime);
+    test_suse_based_image(image => $derived_image, runtime => $runtime);
+}
+
+sub test_suse_based_image {
+    my %args   = @_;
+    my $distri = $args{distri} //= get_required_var("DISTRI");
+    $distri eq 'sle' ? test_sle_based_image(%args) : test_opensuse_based_image(%args);
```

Comment:
Shouldn't this differentiate based on whether the passed `$args{image}` is openSUSE or SLE? If so, `$args{image} ~= s/opensuse/` or so might do the trick

Reply:
And if not, using `get_required_var("DISTRI");` doesn't really make much sense

Reply:
No. As it is for now this is not needed. all the logic for test running in osd is on `test_sle_based_image` whatever the host. if the test is running in O3 no change in the flow( it still does what it used to do until now) and as for now `get_suse_container_urls` returns public images.

Reply:
It seems like `$args{distri}` is never set, so it could be removed and just use `is_sle`.

Reply:
this line is not from me. and distri parameter seems it is not used but i cant see the problem as it is set by `get_required_var` by default.

Reply:
There is no problem per se, but it's still dead code. However, you could just use ```test_suse_based_image(image => $derived_image, runtime => $runtime, distri => $distri);``` in line 90 and it would no longer be.

Reply:
i see your point but this seems to be out of the scope of this PR. i havent dont any change in the flow as it was before. `build_with_zypper_docker` is not affected by `test_suse_based_image` which was introduced just to simplify the workflow of the `test_opensuse_based_image`. i think if i go with `test_suse_based_image(image => $derived_image, runtime => $runtime, distri => $distri);` in line 90 i would have to refactor again many other test modules. i do not know the reason it was implemented like that in first place so i would rather do a refactoring and improvements in another PR? unless i am missing something.

Reply:
> i havent dont any change in the flow as it was before.

Previously, `build_with_zypper_docker` called `test_opensuse_based_image` directly, but now it's calling either `test_sle_based_image` or `test_opensuse_based_image` based on `get_required_var("DISTRI");`.

`build_with_zypper_docker` supports overriding `$distri`, but does not pass this down to `test_suse_based_image`. This means that if `build_with_zypper_docker` is called for an openSUSE image on SLE, it would break. AFAICT the `$distri` override is meant to allow exactly that?

Reply:
From my POV this is unlikely due to `get_suse_container_urls`. and the `build_with_zypper_docker` is not called when `can_build_sle_base($iname)` is false. Sorry i cant see the bug

Reply:
The `build_with_zypper_docker` should be called when `can_build_sle_base` is false as well as `zypper-docker` supports openSUSE too.